### DISCUSSION TITLE
Unexpected token '{' in expression or statement

### DISCUSSION THREAD

**Discussion by Joezczhang** [PowerShell/PowerShell#17137] [General]
### Prerequisites

- [x] Write a descriptive title.
- [ ] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [x] Search the existing issues.
- [ ] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [ ] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

I am writing a scrip that auto login, i got a example in https://github.com/PowerShell/PowerShell/blob/7dc4587014bfa22919c933607bf564f0ba53db2e/test/powershell/Host/assets/Read-Host.Output.expect.
but there is something error just like below:
Line |  "prompt: $" { send "input\r" ; exp_continue }
  10 |     Unexpected token '{' in expression or statement.

### Expected behavior

```console
this scrip should exit with 0
```

### Actual behavior

```console
Unexpected token '{' in expression or statement.
```

### Error details

```console
Unexpected token '{' in expression or statement.
```

### Environment data

```powershell
powershell-7.2.2-win-x64
```

### Visuals

![image](https://user-images.githubusercontent.com/89260661/162612877-2746881b-6be1-44ef-9f16-2a53a5e487b7.png)

**Comment by kasini3000** [comment]
are you looking for expect powershell version? i write one 
https://gitee.com/chuanjiao10/kasini3000/blob/master/k_run_ip_port_w2l_user_expect_pwd.ps1


<blockquote><img src="https://gitee.com/static/images/logo_themecolor.png" width="48" align="right"><div>Gitee</div><div><strong><a href="http://gitee.com/chuanjiao10/kasini3000/blob/master/k_run_ip_port_w2l_user_expect_pwd.ps1">k_run_ip_port_w2l_user_expect_pwd.ps1 · chuanjiao10/kasini3000 - Gitee.com</a></strong></div><div>类似于Puppet，Ansible，pipeline。号称比ansible好10倍，具有批量远程推送脚本，执行脚本，远程传递代码块，定时执行的，devops运维工具。</div></blockquote>

**Comment by Joezczhang** [comment]
Yse, I am searching solution like expect in linux.
i want to do adb shell with account is a, password is b.
I finished it by vbs,like below

Dim WshShell
Set WshShell=WScrip.CreateObject(“WScrip.Shell”)
WshShell.Run "cmd.exe"
WshShell.SendKeys "adb shell"
WScrip.Sleep 2000
WshShell.SendKeys "a"
WScrip.Sleep 2000
WshShell.SendKeys "b"
WScrip.Sleep 2000

is there a simple solution?

i can't get https://www.powershellgallery.com/packages/Await/0.8 by internet, the link does not work.
hope for more information 

<blockquote><img src="https://powershellgallery.com/Content/Images/Branding/PackageDefaultIcon.png" width="48" align="right"><div><strong><a href="https://powershellgallery.com/packages/Await/">Await 0.8</a></strong></div><div>Await - A modern implementation of EXPECT for Windows. For a demo, see: https://www.youtube.com/watch?v=tKyAVm7bXcQ</div></blockquote>

**Comment by vexx32** [comment]
This doesn't look like an issue with pwsh itself, so I will convert this to a discussion for the time being. 🙂
